src/components/selector_menu/index.js

```diff
@@ -63,48 +65,57 @@ var SelectorMenu = React.createClass({
     }
   },
 
-  getOptionNames: function() {
-    // Returns a list of option names, plus the default value.
-    var options = [this.props.selection || this.state.selected || this.props.defaultSelection];
-
-    _.each(this.props.optionsList, function(option) {
-      return option.title ? options.push(option.title) : options.push(option.name);
+  /*
+   * Returns a list of option names, plus the default value.
+   */
+  getOptionNames() {
+    let options = [this.props.selection || this.state.selected || this.props.defaultSelection];
+
+    this.props.optionsList.forEach((option) => {
+      let attr = option.title ? 'title' : 'name';
+      if (!options.includes(option[attr])) {
```

Comment:
I just came across this the other day, but do you know about the new Set type? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

I think this turns into:
   
```js
let mySet = new Set(pluck(this.props.optionsList.map( (opt) => ( opt.title || opt.name ) ));
mySet.add(this.props.selection || this.state.selected || this.props.defaultSelection);
```